src/engine/effects/engineeffectchain.cpp

```diff
@@ -23,6 +23,8 @@ EngineEffectChain::EngineEffectChain(const QString& group,
         }
         m_chainStatusForChannelMatrix.insert(inputChannel.handle(), outputChannelMap);
     }
+
+    m_effectsDelay = new EngineEffectsDelay();
```

Comment:
No need for heap allocation here imo. And if there is prefer unique_ptr. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es60-avoid-new-and-delete-outside-resource-management-functions

Reply:
Can you still explain why you are doing heap-allocation here? I don't think that is necessary.